ptp/ptp4u/server/server_test.go

```diff
@@ -121,3 +122,28 @@ func TestSendGrant(t *testing.T) {
 
 	s.sendGrant(sc, &ptp.Signaling{}, 0, 0, 0, sa)
 }
+
+func TestDrain(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	s := Server{
+		ctx:    ctx,
+		cancel: cancel,
+	}
+
+	require.NoError(t, s.ctx.Err())
+	s.Drain()
+	require.Error(t, s.ctx.Err())
```

Comment:
Do you know exact error? It would be ideal if we check `ErrorIs()`

Reply:
Oh, it is public, maybe we can. let me check.

Reply:
Added